### ISSUE TITLE
WebHook call not received when a change is made to subscription in Azure SaaS Dashboard

### ISSUE THREAD

**Issue by BayScallop** [Azure/Commercial-Marketplace-SaaS-Accelerator#692]
I don't know if you can help and I see this is closed, but I am having an issue where the (Azure Marketplace) WebHook I created never receives a notification from the Azure Marketplace. I have followed the patterns in the docs and can hit the webhook API's with Postman and monitor the WebHook in the Azure Logs. The WebHook is an Azure SaaS service. I have opened a case with Microsoft Partner Center but they cannot seem to provide any answers. This is the last item I need to complete before submitting my offer for certification. I can provision and activate a test subscription from the Landing Page but when I make changes in the Azure SaaS Dashboard, no webhook call occurs. I figure it is either: 1) blocked at the network level (don't think so as it is public); 2) Saas service is not in a correct state to send WebHook messages (I can activate a test subscription); 3) Issue with endpoint (I have tired several as the docs and online postings have several variations) and I don't see any attempts to any endpoint - generally a bad endpoint shows a 404 in the logs. So the most likely is item 2 unless I am totally missing something.

**Comment by BayScallop** [comment]
Where would I find the incident number?

I believe I has resolved this issue.

I do have  a follow on question: I would like to have the Fulfillment Landing Page and the Fulfillment WebHook to be distinct applications to preserve "Seperation of concerns". Can that be done? the SaaS Accelerator show both built into a single service.

Thanks!

Stephen (Steve) Wheeler | Wheeler
Technologies, Inc.
President
Phone:
813.818.4406 (office)
Phone:
954.612.5197 (mobile)
Email:  ***@***.***
 
"Certified Service Disabled Veteran Owned Small Business"
 
PLEASE
READ: The material contained in this email is confidential and solely for the
use of the intended recipient. Please do not forward this email to others without
permissions. All email sent to or from this address will be recorded and
archived by Wheeler Technologies, Inc. (WTI) corporate email system and is
subject to archival, monitoring or review by, and/or disclosure to, someone
other than the recipient. WTI's privacy policy can be found at https://www.wheelertech.us/privacy
 
WARNING:
For your own protection, avoid sending identifying information such as social
security or account numbers to us, or others via email.
 

________________________________________

**Comment by santhoshb-msft** [comment]
Hi @BayScallop That is true, both are in the same app. You can absolutely separate it but that would be your custom implementation..

**Comment by BayScallop** [comment]
Yes, I prefer to separate the 2 distinct functions into a WebHook API and a LandingPage UX, each having a distinct App Id in Entra App Registrations.

In the Technical Configuration for offers, there is a place for a TenantId and an Application Id.

Would the Application Id be for the WekHook App Id or the Landing Page Id? Or can you have multiple App Id's?

The trouble I was having was that I was not getting calls on my WebHook API when I made changes to a test subscription.

I wondered if they could be separate since the SaaS Accelerator they are in a single App.

I now suspect that the problem I had was due to a bad URL and not the fact the Web Hook and Landing Page were each distinct apps.

So, the App Id in the Tech configuration page is that for the Web Hook or the Landing Page? Can you have multiple App Id's?

I suspect the App Id is used for token generation but both the Web Hook and the Landing Page get tokens.

Please help me to understand this better.

Sincerely,

Stephen (Steve) Wheeler | Wheeler
Technologies, Inc.
President
Phone:
813.818.4406 (office)
Phone:
954.612.5197 (mobile)
Email:  ***@***.***
 
"Certified Service Disabled Veteran Owned Small Business"
 
PLEASE
READ: The material contained in this email is confidential and solely for the
use of the intended recipient. Please do not forward this email to others without
permissions. All email sent to or from this address will be recorded and
archived by Wheeler Technologies, Inc. (WTI) corporate email system and is
subject to archival, monitoring or review by, and/or disclosure to, someone
other than the recipient. WTI's privacy policy can be found at https://www.wheelertech.us/privacy
 
WARNING:
For your own protection, avoid sending identifying information such as social
security or account numbers to us, or others via email.
 

________________________________________
